CONTRIBUTING.rst

```diff
@@ -3,15 +3,48 @@ Contributing to PySCIPOpt
 
 Code contributions are very welcome and should comply to a few rules:
 
-1. Compatibility with both Python-2 and Python-3
+0. Read `Design principles of PySCIPOpt`_.
 
-#. All tests defined in the Continuous Integration setup need to pass:
+1. Compatibility with both Python-2 and Python-3.
+
+2. All tests defined in the Continuous Integration setup need to pass:
 
     - `.travis.yml <.travis.yml>`__
     - `appveyor.yml <appveyor.yml>`__
 
-#. New features should be covered by the tests, so please extend `tests <tests>`__
+3. New features should be covered by tests *and* examples. Please extend `tests <tests>`__ and `examples <examples>`__. Tests uses pytest and examples are meant to be accessible for PySCIPOpt newcomers (even advanced examples).
```

Comment:
Hi

I do not know if it is clear here what "feature" means. In point 6 a distinction is made between different types of features is given. We certainly do not require tests for wrapping basic SCIP code. In the design principles is the first place where convenience functions are mentioned which are the functions that definitely need tests an examples

So I think it would make more sense to try to explicit this further. In particular, the non-optionality of test and examples for convenience functions (I know @mattmilten suggested to remove that sentence, but I think that it is ok if it is explicitly mentioned that it is for convenience functions)

Reply:
I say, we don't put it in as mandatory and just hope that people add tests. If not we can always just ask for (more) tests in the PR.

Reply:
I look for suggestions about "features": maybe it is just wording, maybe it is not... I used the term "feature" from the discussion in Berlin.

About testing: are you sure we do not need tests for pure wrapping functions? I think these functions have an pretty defined expected behaviour (SCIP!), therefore it must be tested to comply with that behaviour.

Reply:
I agree with @fserra that simple wrapper functions do not require testing. We should assume that SCIP works correctly and only test those parts that PySCIPOpt adds on top.

Reply:
The assumption that SCIP works correctly does not ensure PySCIPOpt pure wrapper to works as correctly as SCIP. Right?

Reply:
True, I suppose you can never guarantee that. I still think that we do not need to enforce those tests. If people add them by themselves it's perfectly fine, of course.